packages/angular/src/wallet/wallet.store.ts

```diff
@@ -0,0 +1,295 @@
+    readonly ready$ = this.select((state) => state.ready);
+    readonly anchorWallet$ = this.select(
+        this.publicKey$.pipe(isNotNull),
+        this.adapter$.pipe(isNotNull),
+        (publicKey, adapter) => {
+            return publicKey && 'signTransaction' in adapter && 'signAllTransactions' in adapter
+                ? {
+                      publicKey,
+                      signTransaction: (transaction: Transaction) => this.signTransaction(transaction)?.toPromise(),
+                      signAllTransactions: (transactions: Transaction[]) =>
+                          this.signAllTransactions(transactions)?.toPromise(),
+                  }
+                : undefined;
+        }
+    );
+
+    constructor(
+        @Optional()
+        @Inject(WALLET_CONFIG)
+        private _config: WalletConfig
+    ) {
+        super({
+            wallets: _config.wallets,
+            selectedWallet: null,
+            wallet: null,
+            adapter: null,
+            connected: false,
+            connecting: false,
+            disconnecting: false,
+            ready: false,
+            publicKey: null,
+            autoApprove: false,
+        });
+
+        if (!this._config) {
+            this._config = WALLET_DEFAULT_CONFIG;
+        } else {
+            this._config = {
+                ...WALLET_DEFAULT_CONFIG,
+                ...this._config,
+            };
+        }
+
+        const walletName = localStorage.getItem(this._localStorageKey);
+        const wallet = this._config.wallets.find(({ name }) => name === walletName);
+
+        if (wallet) {
+            this.selectWallet(walletName as WalletName);
+        } else if (this._config.wallets.length > 0) {
+            this.selectWallet(this._config.wallets[0].name);
+        }
+    }
+
+    readonly autoConnect = this.effect(() => {
+        return combineLatest([this.adapter$, this.ready$]).pipe(
+            filter(([adapter, ready]) => this._autoConnect && !!adapter && ready),
+            observeOn(asyncScheduler),
+            tap(() => this.connect())
+        );
+    });
+
+    readonly connect = this.effect((action$: Observable<void>) => {
+        return action$.pipe(
+            concatMap(() => of(null).pipe(withLatestFrom(this.state$, (_, state) => state))),
+            filter(({ connected, connecting, disconnecting }) => !connected && !connecting && !disconnecting),
+            tap(() => this.patchState({ connecting: true })),
+            concatMap(({ adapter, wallet, ready }) => {
+                if (!wallet || !adapter) {
+                    this.logError(new WalletNotSelectedError());
+                    return of(null);
+                }
+
+                if (!ready) {
+                    window.open(wallet.url, '_blank');
+                    this.logError(new WalletNotReadyError());
+                    return of(null);
+                }
+
+                return from(defer(() => adapter.connect())).pipe(catchError(() => of(null)));
```

Comment:
My reading of this is that it's doing something like
```ts
try {
    await adapter.connect();
} catch (error) {
    return null;
}
```
Is that accurate? It doesn't seem to be logging or rethrowing the error.

Reply:
Yes, this is accurate. The thing is that I have the adapter.on('error') also running, if there's an error during connection I would expect that to catch it, am I wrong here? 

Let me guess, not all wallets work the same and some do fail during connect but the error is not emitted through the error event.

Reply:
The adapter will always both emit an event and throw an error if the connect call fails, so I think you're right that we don't need to log that, since it will already get logged.

 I'm wondering how it works when you call `connect` in your app. The global error handler is really just there to provide end-user notifications, so you can still catch errors from your `connect` call (or ignore them like https://github.com/solana-labs/wallet-adapter/blob/e94577e73b483c2425006a28a37f9f2662ee12f5/packages/react-ui/src/WalletConnectButton.tsx#L13).

But maybe there's a more idiomatic way of doing this in this framework?

Reply:
This is something peculiar of how the @ngrx/component-store works, when you trigger an effect, you **can't** wait for a response, you just fire and forget. If you take a look at the send and sign methods, I couldn't use an effect because otherwise, consumers wouldn't be able to wait for the response to do something else. 

The idea is to make effects more declarative but that sometimes hurts ergonomics IMHO. My rule of thumb is, do I really need to wait for the response? If I do, I return an observable and let consumers handle the lifecycle of the observable, otherwise, I encapsulate it inside an effect.

Reply:
I see. Can we at least return the type of error that was caught instead of null, or is there another way to access it for consumers?

Reply:
Oh, an important new behavior I missed in review -- https://github.com/solana-labs/wallet-adapter/blob/e94577e73b483c2425006a28a37f9f2662ee12f5/packages/react/src/WalletProvider.tsx#L127-L143

If the connection fails, we reset the state, including the selected wallet. Many dapps just show a single select/connect button, and this prevents the user from getting locked into a state where the wallet is selected but connecting fails (for example, if the wallet isn't viable on their platform). Especially since the state is in localStorage, this produced a really bad UX where the user had now way to recover except to know to clear their cookies.

Reply:
Funnily this method returns void and that's how the library works LOL. But what you usually do, is create an error stream and push everything there, that way consumers can do:

```typescript
error$.subscribe((error) => notifyError)

// or this
error$
  .pipe(filter((error) => error instanceof WhatEverError))
  .subscribe((error) => notifyError)
```